### PR TITLE
[Fix common tests on GPU] send model, ids to torch_device

### LINKED ISSUE #4013: test_lm_head_model_random_*_generate fail on GPU

### PR DESCRIPTION
**Pull request by sshleifer**

### CODE REVIEW

File: tests/test_modeling_common.py [hunk 1]
```diff
@@ -684,28 +689,24 @@ def test_lm_head_model_random_beam_search_generate(self):
...
+            ]
             output_tokens = model.generate(
                 input_ids, do_sample=False, bad_words_ids=bad_words_ids, num_beams=2, num_return_sequences=2
             )
             # only count generated tokens
             generated_ids = output_tokens[:, input_ids.shape[-1] :]
             self.assertFalse(self._check_match_tokens(generated_ids.tolist(), bad_words_ids))
 
-    def _generate_random_bad_tokens(self, num_bad_tokens, model):
+    def _generate_random_bad_tokens(self, num_bad_tokens: int, config: PretrainedConfig) -> list:
```
Comment (julien-c) [inline tests/test_modeling_common.py:703]:
```suggestion
    def _generate_random_bad_tokens(self, num_bad_tokens: int, config: PretrainedConfig) -> List[in]:
```

**Review by julien-c — APPROVED** [review]

**Comment by sshleifer** [comment]
@patrickvonplaten 
 any idea why tf tests dont run? 
https://app.circleci.com/pipelines/github/huggingface/transformers/5734/workflows/7986a344-3a5f-4f4f-a5e7-f8fd82ef7a2d/jobs/33381/steps
seems like test_modeling_common.py only used by torch.

File: tests/test_modeling_common.py (hunk 1 above)
Comment (patrickvonplaten) [reply tests/test_modeling_common.py]: `int`

**Comment by patrickvonplaten** [comment]
Great, thanks for fixing this :-)

**Review by patrickvonplaten — APPROVED** [review]

**Review by LysandreJik — APPROVED** [review]
Cool, LGTM!

**Pull request merged**